packages/ui/rating/rating-modal/RatingModal.tsx

```diff
@@ -1,48 +1,105 @@
 import {
   Button,
-  Title,
-  Rating,
+  DEFAULT_THEME,
+  Divider,
   Flex,
-  Text,
   Image,
-  Divider,
-  DEFAULT_THEME,
+  Rating,
+  Text,
+  Title,
 } from '@mantine/core';
+import { useMediaQuery } from '@mantine/hooks';
+import { ChangeEvent, useState } from 'react';
+
+import { IOrder } from '../../../../apps/web/src/views/user-transaction/types';
 import StarSVG from '../StarSVG';
 import {
   StyledImageWrapper,
+  StyledModal,
   StyledProductDiv,
   StyledTextarea,
-  StyledModal,
+  StyledTitleArea,
 } from './styles';
-import { useMediaQuery } from '@mantine/hooks';
-import { useState } from 'react';
-
 interface RatingModalProps {
+  onRatingSubmit: (rating: RatingData) => void;
   isOpen: boolean;
   onClose: () => void;
+  data: IOrder;
+  productId: string;
+  closeModal: () => void;
```

Comment:
we can remove this for now to avoid closing of modal when there is incorrect input